### ISSUE TITLE
Notification contents are not shown on Android Pie

### ISSUE THREAD

**Issue by nickaein** [Telegram-FOSS-Team/Telegram-FOSS#303]
The notification shows up but the message content simply says "Message" instead of actual message text:

https://imgur.com/a/Ic1BhWb

This problem is also present in the "official" (DrKLO) Telegram app. 

Other applications (e.g. Gmail app) doesn't have such issue.

Configuration:
Telegram-FOSS version: v5.3.1 (1497)
Android: Pie
Local Passlock: Disabled

**Comment by Bubu** [comment]
Is this a secret chat?

**Comment by nickaein** [comment]
No it's public channel.

The same problem is present for chat groups and private messages (non-secret) too.

**Comment by M86xKC** [comment]
1. Make sure telegram is not locked by  *passcode* (settings > privacy and security > security > passcode lock)

There is an option in the settings to preview the message or not (settings > notifications and sounds > notifications for chats > choose messages you want to be previewed >  turn on message preview.

**Comment by nickaein** [comment]
1. It's disabled.

2. I couldn't find this. Searching for "Preview" in settings doesn't show up anything. Is this available on Android Pie? 

BTW, I haven't done any customization to Telegram notification, neither to global notifications.

**Comment by nickaein** [comment]
I'm might give it try and debug it. It seems https://github.com/Telegram-FOSS-Team/Telegram-FOSS/blob/61b5c143e59bd87c0df7e82e09688aa3aaa5daa3/TMessagesProj/src/main/java/org/telegram/messenger/NotificationsController.java#L2615 and https://github.com/Telegram-FOSS-Team/Telegram-FOSS/blob/61b5c143e59bd87c0df7e82e09688aa3aaa5daa3/TMessagesProj/src/main/java/org/telegram/messenger/NotificationsController.java#L3246 are places that the notification is being created. Maybe some tweaks helps to get it working. 
Your suggestions are also very welcome.

I'm new to Android development though. Should I download **Android NDK 16b** as [build documentation](https://github.com/Telegram-FOSS-Team/Telegram-FOSS#building) says or for targeting Android Pie correctly, a newer NDK would be better?

**Comment by nickaein** [comment]
@Bubu What info is needed? I would be happy to provide them.

**Comment by Bubu** [comment]
@nickaein it seems to only happen for you. So some way to reproduce this would be needed.

**Comment by nickaein** [comment]
@Bubu I see. My initial assumption was that this issue occurs on Android Pie. Is the cases that dismiss this?

**Comment by Bubu** [comment]
I can test on a Pie device soon, thanks for the pointer.

**Comment by Grossdm** [comment]
Hello,
I have a general question to ask. First let me say:
• I'm new to Telegram
• I'm new to Pie (it's OxygenOS fork)

Basically, how should I configure notifications?
I have been directed to the instructions for Oreo, which don't seem to apply.

BTW, I made a big jump in Android versions.
I went from my Paleophone running KitKat to a OnePlus 6T with Pie. Yup, that's version 4 to 9.

Perhaps this gives me less understanding than most users.

Any help would be appreciated.
TIA,
Doug

**Comment by nroach44** [comment]
I'm getting something similar - the device receives message fine (the app has the messages) but the only notification that is visible is the persistent service for push messages (since I'm running without gapps).

Should I chuck that in another ticket?

**Comment by M86xKC** [comment]
@nroach44 did you enable settings > Notifications and Sounds > Other > Background Connection ?

**Comment by nroach44** [comment]
@M86xKC Yes - the app was recieving the messages fine, but wasn't notifying. In this case it was because the behaviour of the "do not disturb" times had changed between phones - so I was in DnD mode, and the phone wasn't showing notifications as a result.

Cheers though!

**Comment by Pannuba** [comment]
@Bubu I'm having this issue but only with secret chats: notifications simply read "You have a new message", without even specifying the sender. Is that intended behaviour or a bug?

**Comment by smnthermes** [comment]
@Pannuba The user can hide the notifications content in the lockscreen, but not when the phone is unlocked, which is the ideal behavior. However, Telegram hides the content even when the phone is unlocked.

**Comment by nathyharumi** [comment]
Hi!

I was having the same issue and then, when I clicked at notifications for chats > private chats, it showed a lot of options, including message preview. It was disabled, I don't know why.

**Comment by rohantmp** [comment]
Same issue.
- Custom for one chat (with my own telegram-send bot)
- Message Preview Enabled.
- Tried with every importance.

**Comment by jorgechurriana** [comment]
> Hi!
> 
> I was having the same issue and then, when I clicked at notifications for chats > private chats, it showed a lot of options, including message preview. It was disabled, I don't know why.

This work for me too, thanks

**Comment by Swissnode** [comment]
> 
> 
> Hi!
> 
> I was having the same issue and then, when I clicked at notifications for chats > private chats, it showed a lot of options, including message preview. It was disabled, I don't know why.

man you saved me, was looking for days! I didn't realise that those options are actually clickable and then release a whole subset of options. Your message saved my bacon!! How those setting were changed is beyond me though!
